datasm/datasm.py

```diff
@@ -515,25 +519,31 @@ def start_datasets(self, datasets=None):
                 log_message("info", f"start_datasets: calling setup_requisites : {source_dataset.dataset_id}")
                 job.setup_requisites(source_dataset)
                 job_reqs_met = job.meets_requirements()
-                log_message("info", f"start_datasets: job_reqs_met={job_reqs_met}")
+                log_message("info", f"start_datasets: job_reqs_met={job_reqs_met}") # True/False
             if job.job_id is None and job_reqs_met:
-                log_message("info", f"start_datasets: Job {job_name} meets its input dataset requirements")
-                job_id = job(self.slurm)
+                log_message("info", f"start_datasets: (job.job_id={job.job_id}) Job {job_name} meets its input dataset requirements, calling job(self.slurm)")
+                job_id = job(self.slurm)        # WARNING:  If job script exits prematurely, this call never returns and datasm hangs.
                 log_message("info", f"start_datasets: DGB: got job_id {job_id} from job(self.slurm)")
                 if job_id is not None:
+                    log_message("info",f"Adding job with job_id {job_id} to self.job_pool")
                     job.job_id = job_id
                     self.job_pool.append(job)
                 else:
+                    log_message("info", f"Error starting up job {job}")
                     log_message("error", f"Error starting up job {job}")
+                    continue
             else:
                 log_message("error", "DGB: job NOT added to pool")
                 log_message("info", f"start_datasets: DGB: job.job_id = {job.job_id}, job_reqs_met = {job_reqs_met}")
                 attributes = [attr for attr in dir(job) if not attr.startswith('__')]
                 for attr in attributes:
                     print(f"{attr} = {getattr(job,attr)}")
             log_message("info", f"start_datasets: (bottom loop: for job in new_jobs)")
-        log_message("info", f"start_datasets: Return")
-        log_message("info", f"start_datasets: (hangs here until slurmed workflows complete)")
+        job_pool_size = len(self.job_pool)
+        if len(self.job_pool) == 0:
+            log_message("info", f"Job Pool contains No jobs, pass={zpasses}:  datasm NOT exiting.")
+            # sys.exit(0)
```

Comment:
```suggestion
```

Reply:
I can do that.  Anything that says "ipdb.settrace()" was Sterling's work - I was never very good at runtime-debugger ops.